c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -142,7 +147,13 @@ impl<'tcx> TypeChecker<'tcx, '_> {
             Rvalue::Len(pl) => {
                 self.visit_place(pl, Mutability::Not);
             }
-            Rvalue::Cast(_, ref op, _) => self.visit_operand(op),
+            Rvalue::Cast(_cast_kind, ref op, _) => {
+                // A cast such as T as U
+                let t_lty = self.acx.type_of(op);
+                let u_lty = rvalue_lty;
+                assert!(is_castable_to(t_lty, u_lty));
+                self.visit_operand(op)
+            }
```

Comment:
I think something like `from` and `to`, or `pl` and `rv` are clearer than `t` and `u`.

Reply:
https://github.com/immunant/c2rust/pull/883/commits/d9a6de02ce88087a4cc60a239122696907863d87